### PR TITLE
chore: Use .localhost instead of .local 🦭

### PR DESCRIPTION
**Pull request by darcywong00**
This updates follows the latest guidance to use .localhost instead of .local on Devs' locally hosted sites

Mirrors keymanapp/help.keyman.com#801

### CODE REVIEW

**Review by mcdurdin — APPROVED** [review]
LGTM

**Comment by darcywong00** [comment]
Troubleshooting the failing CI test

Seems the khmer search uses a weight 245 for sil_khmer but the expected weight 356.
`"tag": "km"` is also in a different order

**Comment by mcdurdin** [comment]
> Seems the khmer search uses a weight 245 for sil_khmer but the expected weight 356.

hmm, the datasets are supposed to be stable for the unit tests so that's strange.

File: _common/KeymanHosts.php [hunk 1]
```diff
@@ -107,12 +109,29 @@ function __construct() {
         $this->keyman_com = "https://keyman-staging.com";
         $this->keymanweb_com = "https://keymanweb.com";
         $this->r_keymanweb_com = "https://r.keymanweb.com";
+      } else if($this->tier == KeymanHosts::TIER_TEST) {
+        $this->s_keyman_com = "https://s.keyman.com";
+        $this->api_keyman_com = "http://host.docker.internal:8098"; // Unique for api.keyman.com
```
Comment (mcdurdin) [inline _common/KeymanHosts.php:114]: btw, why 8098 and not 8058?

**Review by mcdurdin — COMMENTED** [review]
> > Seems the khmer search uses a weight 245 for sil_khmer but the expected weight 356.
>
> hmm, the datasets are supposed to be stable for the unit tests so that's strange.

This smells like we are no longer testing against the static test data. I wonder if the changes to hosts mean that the env is no longer TIER_TEST or something like that?

File: _common/KeymanHosts.php (hunk 1 above)
Comment (mcdurdin) [reply _common/KeymanHosts.php:114]: That is, all the other sites are in the range 805x. Only api.keyman.com differs.

**Comment by mcdurdin** [comment]
https://github.com/keymanapp/api.keyman.com/actions/runs/6009846050/job/16300063487?pr=206#step:4:1765

```
   │              "sourcePath": "release/k/khmer_angkor",
   │ -            "url": "https://keyman-staging.com/go/package/download/khmer_angkor?version=1.0.6&platform=windows&tier=alpha&update=0",
   │ +            "url": "https://keyman.com/go/package/download/khmer_angkor?version=1.0.6&platform=windows&tier=alpha&update=0",
   │              "version": "1.0.6"
```

This shows that the test is expecting keyman-staging.com (and help.keyman-staging.com) in KeymanHosts for TIER_TEST.

File: _common/KeymanHosts.php (hunk 1 above)
Comment (darcywong00) [reply _common/KeymanHosts.php:114]: It's just ports we ended up on after changing several things during EMDC.
I can try 8098 and 8099

File: _common/KeymanHosts.php [hunk 2]
```diff
@@ -107,12 +109,29 @@ function __construct() {
         $this->keyman_com = "https://keyman-staging.com";
         $this->keymanweb_com = "https://keymanweb.com";
         $this->r_keymanweb_com = "https://r.keymanweb.com";
+      } else if($this->tier == KeymanHosts::TIER_TEST) {
+        $this->s_keyman_com = "https://s.keyman.com";
+        $this->api_keyman_com = "http://host.docker.internal:8058"; // Unique for api.keyman.com
```
Comment (darcywong00) [inline _common/KeymanHosts.php:114]: > This smells like we are no longer testing against the static test data. I wonder if the changes to hosts mean that the env is no longer TIER_TEST or something like that?

Ah. I see the original TIER_TEST used `https://api.keyman-staging.com` and not the actual repo?
https://github.com/keymanapp/api.keyman.com/blob/60a7ab4f0b4da5d341ac90606f71d90e2b58db64/_common/KeymanHosts.php#L100-L104

Comment (mcdurdin) [reply _common/KeymanHosts.php:114]: > Ah. I see the original TIER_TEST used `https://api.keyman-staging.com` and not the actual repo?

TIER_TEST uses local test data, so we can ensure that we get consistent results.

Comment (darcywong00) [reply _common/KeymanHosts.php:114]: I'm still confused.

> $this->api_keyman_com = "https://api.keyman-staging.com";

Doesn't this mean the live staging branch ci would check api.keyman-staging.com for its tests?
My change was setting to `http://host.docker.internal:8058`, which I thought would be local test data.

File: _common/KeymanHosts.php [hunk 3]
```diff
@@ -107,12 +109,29 @@ function __construct() {
...
+        $this->help_keyman_com = "https://help.keyman.com";
+        $this->downloads_keyman_com = "https://downloads.keyman.com";
+        $this->keyman_com = "https://keyman.com";
```
Comment (mcdurdin) [inline _common/KeymanHosts.php:117]:
```suggestion
        $this->help_keyman_com = "https://help.keyman-staging.com";
        $this->downloads_keyman_com = "https://downloads.keyman.com";
        $this->keyman_com = "https://keyman-staging.com";
```

File: _common/KeymanHosts.php (hunk 2 above)
Comment (mcdurdin) [reply _common/KeymanHosts.php:114]: > Seems the khmer search uses a weight 245 for sil_khmer but the expected weight 356.

This test seems to be passing now, so I think the only problem left is the test failure I highlighted earlier:

> https://github.com/keymanapp/api.keyman.com/actions/runs/6009846050/job/16300063487?pr=206#step:4:1765
> 
> ```
>    │              "sourcePath": "release/k/khmer_angkor",
>    │ -            "url": "https://keyman-staging.com/go/package/download/khmer_angkor?version=1.0.6&platform=windows&tier=alpha&update=0",
>    │ +            "url": "https://keyman.com/go/package/download/khmer_angkor?version=1.0.6&platform=windows&tier=alpha&update=0",
>    │              "version": "1.0.6"
> ```
> 
> This shows that the test is expecting keyman-staging.com (and help.keyman-staging.com) in KeymanHosts for TIER_TEST.

You wrote:
> Doesn't this mean the live staging branch ci would check api.keyman-staging.com for its tests?

Yes, using host.docker.internal is correct. But we can see:

https://github.com/keymanapp/api.keyman.com/blob/d7181388a4f4ea205c0061c41e278d168f83539f/script/windows/14.0/update/WindowsUpdateCheck.php#L190-L195

This is still picking up the wrong answer; I think we need it to be keyman-staging for TIER_TEST.

**Pull request merged**